### ISSUE TITLE
Nuxt extremely slow on initial request

### ISSUE THREAD

**Issue by kirkbushell** [nuxt/nuxt#20596]
### Environment

- Operating System: Darwin
- Node Version:     v16.19.1
- Nuxt Version:     3.4.3
- Nitro Version:    2.3.3
- Package Manager:  npm@8.19.3
- Builder:          vite
- User Config:      css, modules, postcss, rootDir, runtimeConfig, vuefire
- Runtime Modules:  nuxt-vitest@0.6.10, nuxt-vuefire@0.1.6, @pinia/nuxt@0.4.9
- Build Modules:    -

### Reproduction

Start a nuxt project, run npm run dev and then open the site in the browser, now go make a coffee.

### Describe the bug

After running: npm run dev, and opening the site in the browser, that initial request takes 2-3 minutes. After that, refreshes are instant.

**Comment by warflash** [comment]
I guess 2 minutes for really large projects on low performance machines is nothing unreasonable? Depends a lot on your exact setup and project so you might want to provide a reproduction project + specs if you feel like you're running into a bug of sorts.
IIrc docker on mac os has some issues but that's not limited to nuxt, also https://github.com/nuxt/nuxt/pull/20179 is something you might want to check out as that PR specifically adresses dev startup performance

**Comment by danielroe** [comment]
Would you provide a [reproduction](https://nuxt.com/docs/community/reporting-bugs/#create-a-minimal-reproduction)? 🙏

**Comment by kirkbushell** [comment]
> I guess 2 minutes for really large projects on low performance machines is nothing unreasonable? Depends a lot on your exact setup and project so you might want to provide a reproduction project + specs if you feel like you're running into a bug of sorts. IIrc docker on mac os has some issues but that's not limited to nuxt, also #20179 is something you might want to check out as that PR specifically adresses dev startup performance

Assuming I had a low-performance machine, sure.

I guess the question is WHY is it doing this?

**Comment by danielroe** [comment]
A slower first request is typically due to vite performing on-demand transpilation of the assets needed to render a page. (This can be client-side, not just SSR.)

But 2-3 minutes is not expected behaviour. Which is one reason why I asked for a reproduction.

**Comment by kirkbushell** [comment]
@danielroe yeah fair enough. I don't see it on native vue apps, using vite, so not sure.

I'll setup a repro.

**Comment by glennovische** [comment]
Having the same issue, after building to our test env the first initial load is like SUPER slow....

**Comment by danielroe** [comment]
@glennovische Could you open a new issue with a reproduction? This issue is about dev server initial request performance.

**Comment by warflash** [comment]
@glennovische This issue in particular is for dev env slow initial requests. Things like vites initial transpilation don't apply to a built app. I'd probably recommend you open a new issue for built apps alongside a reproduction😊

**Comment by tobychidi** [comment]
I just started experiencing this after pulling up a Nuxt3 version rc12 or something in a new " supposedly better" pc. I have now updated all dependencies but pnpm run dev takes minutes to run. Why could this be happening? to fix this issue I have restarted several times, deleted and reinstalled with node_modules no fix yet. My project is almost empty and I am still troubleshooting. Will start a fresh nuxt project and see if I have the same problem. 


Current output:

Vite warmed up in : 32235ms
Nitro built in 165784ms

🙆‍♂️

**Comment by tobychidi** [comment]
Oh. Wow. Brand new nuxt project
```
Nuxi 3.5.1                                                                                                                                                                                             14:26:06  
Nuxt 3.5.1 with Nitro 2.4.1                                                                                                                                                                            14:26:06
                                                                                                                                                                                                       14:26:12
  > Local:    http://localhost:3000/ 
  > Network:  http://192.168.73.17:3000/

ℹ Vite client warmed up in 16534ms                                                                                                                                                                     14:26:34  
✔ Nitro built in 105166 ms                                                                                                                                                                       nitro 14:28:08
✔ Vite server hmr 19 files in 3808.515ms       
```

This is unexpected.

**Comment by tobychidi** [comment]
Oh Wow. That was it. 😅

**Comment by kirkbushell** [comment]
Happening on mac as well.

**Comment by dickofn** [comment]
> If you are hitting this on Windows it's a pretty painful regression from 3.5.1. You can set `experimental.watcher` to `chokidar` to resolve. That is likely a different issue from the one reported here.
> 
> The fact that you were experiencing this problem on an old project suggests that there is an issue in a dependency which has caused the regression.

how and where to set these settings @danielroe ?

**Comment by tobychidi** [comment]
> > If you are hitting this on Windows it's a pretty painful regression from 3.5.1. You can set `experimental.watcher` to `chokidar` to resolve. That is likely a different issue from the one reported here.
> > The fact that you were experiencing this problem on an old project suggests that there is an issue in a dependency which has caused the regression.
> 
> how and where to set these settings @danielroe ?

In nuxt.config.ts like this

```
export default defineNuxtConfig({
   experimental: {
      watcher: "chokidar",
   },
})
```

**Comment by dickofn** [comment]
can confirm this fixes my slow start problem

```
export default defineNuxtConfig({
   experimental: {
      watcher: "chokidar",
   },
})
```

**Comment by HugeLetters** [comment]
> 

+1
My start-up time on a fresh project went from 1+ minutes to ~5 seconds.

**Comment by sonofmagic** [comment]
After I add this config to `nuxt.config.ts`, Vite client warmed up still very very slow.


repo url: [Atinux/content-wind/demo](https://github.com/Atinux/content-wind/tree/main/.demo) 


Operating System: Win10
Node Version: v16.14.2
Nuxt Version: 3.5.1
Nitro Version: 2.4.1
Package Manager: pnpm@8.5.1
Builder: vite

![image](https://github.com/nuxt/nuxt/assets/15621541/133f6d3b-4769-4f40-be70-6795c66d9007)

`nuxt.config.ts`:

```ts
export default defineNuxtConfig({
  extends: 'content-wind',
  experimental: {
    watcher: "chokidar",
  },
})
```

**Comment by kirkbushell** [comment]
176s tonight. Even worse after the patch.

**Comment by kirkbushell** [comment]
Oh. This is coming to 3.5.3, not 3.5.2?

**Comment by warflash** [comment]
Yes. You can try opting into the edge channel to check and verify the PR daniel merged yesterday resolves your issue if you don't want to wait until 3.5.3. -> https://nuxt.com/docs/guide/going-further/edge-channel#opting-into-the-edge-channel

**Comment by danielroe** [comment]
@danielvodnansky Would you provide a reproduction? Are you using Windows?

**Comment by kirkbushell** [comment]
> After the last upgrade to 3.5.3, I have this:
> 
> i Vite client warmed up in 575035ms √ Nitro built in 557532 ms
> 
> If I remove the bootstrap 5 library, it reduces a lot, but I cannot do this and I don't think the problem is there.

Oh wow... wth

**Comment by kirkbushell** [comment]
> @danielvodnansky Would you provide a reproduction? Are you using Windows? Note that there are two alternative watch implementations, and the one likely to be most performant on windows is `experimental: { watcher: 'parcel' }` which does require also installing `@parcel/watcher` and `watchman`.

This isn't a win-only problem, though. Myself and others have reported similar issues on mac.

I did also upgrade to 3.5.3 with no change.

**Comment by danielroe** [comment]
The issue is that 'slowness' can come from several causes.

1) watcher initialisation. This is a problem particularly unique to windows due to how fs watching is handled in chokidar. And this was my interpretation of what was being reported. (incorrectly)

2) vite warmup or plugin transform issues - this is likely your issue. But I can't investigate without a reproduction.

The best thing at this point (given that this issue has been used to track the first issue) is to open a new issue with a reproduction so we can investigate. 🙏

**Comment by danielvodnansky** [comment]
es, I use windows.

Nuxt config file: 
![code](https://github.com/nuxt/nuxt/assets/35817567/fc962779-4d90-4353-b5ac-ee3f5185e910)

Package.json 
![code2](https://github.com/nuxt/nuxt/assets/35817567/42771dd8-95d2-4d2a-b840-e44df0ea09a2)

**Comment by danielroe** [comment]
Note that to benefit from the speedup in the latest release you need to ensure you have not overridden the value of `experimental.watcher`.

You can also try using the `parcel` watcher. (Install `@parcel/watcher` and then follow steps to install Watchman.)

If you are encountering an issue associated - with CSS, note we are currently tracking an issue associated with vite and scss.

**Comment by CharlesBT** [comment]
It doesn't seems to be solved on 3.5.3 same starting time as previous versions on windows

**Comment by davidivkovic** [comment]
Same issue here on a fresh windows project. No libraries or anything. Nuxt warmup takes 36 seconds.

**Comment by RomanSkrypnik** [comment]
Same. It takes quite enough time to build my project...

 ├─ /api/_content/cache.1689611793221.json (106544ms)                                                                                                                                                nitro 7:40:12 PM
  ├─ /es/sitemap.xml (109180ms)                                                                                                                                                                       nitro 7:40:14 PM
  ├─ /sitemap.xml (110086ms) 
  
  
  
<img width="556" alt="image" src="https://github.com/nuxt/nuxt/assets/44501848/f0ae1c27-f266-4e74-bf92-ab672255c467">
<img width="574" alt="image" src="https://github.com/nuxt/nuxt/assets/44501848/13605aae-4374-4e2a-b5be-70365df251e2">

**Comment by kirkbushell** [comment]
I've moved on from nuxt, because of this problem - it is so frustrating and wastes so much time when you have to restart the server.

Real shame.

**Comment by ks-neerajn** [comment]
Is any update on load speed of nuxt 3, because I faced this issue one week back, @danielroe?

**Comment by earl-grey-guy** [comment]
This worked for me (as suggested above). My build is down from 20s to 2s

- Upgraded to Nuxt 3.6.5 (this makes some improvement on its own)
- Install Watchman using [windows binary method](https://facebook.github.io/watchman/docs/install). Make sure it's on your PATH via `watchman --version`
- `npm i --save-dev @parcel/watcher` (my version is 2.2.0)
- Update nuxt config as below:

```
export default defineNuxtConfig({
  ...(other stuff),
  experimental: {
    watcher: 'parcel'
  }
})
```

**Comment by danielroe** [comment]
@stephansteiner Are you using layers, or a monorepo? Would you test on the [edge channel](https://nuxt.com/docs/guide/going-further/edge-channel)?

**Comment by earl-grey-guy** [comment]
@stephansteiner  I don't think my solution requires either python or vscode.

**Comment by stephansteiner** [comment]
> @stephansteiner I don't think my solution requires either python or visual studio.

I'm having this issue when installing @parcel/watcher: https://github.com/parcel-bundler/parcel/issues/8152: apparently, unser windows python and visual studio is required as it is used with node-gyp

**Comment by earl-grey-guy** [comment]
@stephansteiner I can't speak to parcel. I used watchman with success; they have a binary so no compilation (python or gyp needed).

**Comment by earl-grey-guy** [comment]
@stephansteiner All I can say is I didn't have that problem when I installed `@parcel/watcher`.  I think the obstacle you're hitting is not nuxt-related at this point. I am sorry you're dealing with it, of course. Node + windows has never been easy mode unfortunately. It's a lot better than it was years ago, but it still kinda sucks.

**Comment by adiramardiani** [comment]
Is any update on this issue ?
Btw my development experience is so bad, I dont know this related to nuxt content, tailwind, or any other thing
But currently my startup time is CRAZY slow, any one experience the same thing again ?
My env is WSL try chokidar, parcel and is same :(

![image](https://github.com/nuxt/nuxt/assets/8703419/8fe64b2f-91f1-42ba-9bc7-bc161f916422)

Just wait 3 minute for nitro built :(

**Comment by danielroe** [comment]
@adiramardiani Would you open a new issue with a reproduction? 🙏

**Comment by pi0** [comment]
It is likely that in windows something is being wrongly bundling in dev mode. Reproduction would greatly help to resolve relevant issues 🙏

**Comment by khalidzamer** [comment]
I have a similar issue, in dev mode, the app loads , but it takes another minute or so in the browser to load color mode switcher in the header and for Pinia to become available, meanwhile, some components are available on the page but not interactive I suspect it has to do with anything ClientOnly or Pinia in a project, even if the project is a simple page with a simple layout it is frustrating I hope the issue is analyzed

**Comment by danielroe** [comment]
@khalidzamer We can investigate if you are able to provide a reproduction.

**Comment by khalidzamer** [comment]
@danielroe , my hero, while I was establishing a new repo that I can share publicly, I tried adding the same packages and configs gradually.. I might have discovered what is causing the issue on my side,

I so far suspect the issue is using `localhost` vs `127.0.0.1` in the browser has an obvious impact,  

`localhost` takes longer to load the color mode switcher, which is loaded a bit delayed after the header, while with `127.0.0.1`  the difference is reduced drastically, 

I tried it in my original working app and the reload time difference is massive and the pain is gone!

Speaking of `localhost` vs. `127.0.0.1` vs `0.0.0.0` , I also notice impact in `/server/api`  requests as well which might behave differently when local vs when deployed, what is your recommendation?

**Comment by pi0** [comment]
@khalidzamer Can you check `nuxi --version`? If it is latest (3.8.3) for windows and WSL it should use `127.0.0.1` by default.

**Comment by khalidzamer** [comment]
@pi0 yes Pooya , that's when I noticed the change in default behavior and suspected something is wrong, although I had this slow behavior last week in previous version, thank you for your efforts

**Comment by stevenhurth** [comment]
Just wondering ... It will take some time for our development team to get used to using 127.0.0.1 for our nuxt apps.

Any chance the issue with using "localhost" on windows will get addressed?

**Comment by pi0** [comment]
@stevenhurth That's true. At that very moment, I had been drafting  https://github.com/unjs/listhen/pull/126. 

So sorry for all frustrations on this matter. As a security measure we had to limit default host and it introduced unforeseen edge cases.

**Comment by stevenhurth** [comment]
> @stevenhurth That's true. At that very moment, I had been drafting [unjs/listhen#126](https://github.com/unjs/listhen/pull/126).
> 
> So sorry for all the frustrations on this matter. As a security measure, we had to limit the default host and it introduced unforeseen edge cases.

No worries at all! Thank you for jumping on it!

**Comment by codetheorist** [comment]
> experimental: {
>       watcher: "chokidar",
>    },

`npm i -g windows-build-tools`

**Comment by randybastian777** [comment]
any update?.
i still have this issues. i cannot use any solutions here.
Windows 11
WSL2 -> Ubuntu 22.04
Nuxt: 3.7.1
------
 WARN  Slow module @nuxt/devtools took 55013.7ms to setup. 
ℹ [nuxt] compiled plugins/client.mjs in 622.37ms 
ℹ Vite client warmed up in 70789ms
✔ Nitro built in 180942 ms

**Comment by 50l3r** [comment]
> @danielroe , my hero, while I was establishing a new repo that I can share publicly, I tried adding the same packages and configs gradually.. I might have discovered what is causing the issue on my side,
> 
> I so far suspect the issue is using `localhost` vs `127.0.0.1` in the browser has an obvious impact,
> 
> `localhost` takes longer to load the color mode switcher, which is loaded a bit delayed after the header, while with `127.0.0.1` the difference is reduced drastically,
> 
> I tried it in my original working app and the reload time difference is massive and the pain is gone!
> 
> Speaking of `localhost` vs. `127.0.0.1` vs `0.0.0.0` , I also notice impact in `/server/api` requests as well which might behave differently when local vs when deployed, what is your recommendation?

I confirm that indeed using 127.0.0.1 instead of localhost has a significant impact:

### 😁 127.0.0.1

https://github.com/nuxt/nuxt/assets/5095592/db2fb73e-1b1f-459a-a0c7-e5f3a79103a9

### 😴 localhost
https://github.com/nuxt/nuxt/assets/5095592/7bf9e8a6-f737-4956-96c6-0b218694f758

**Comment by MGA1600** [comment]
Switching from localhost to 127.0.0.1 did the trick for me.
Using Nuxt 3.7.4. 
Initial page load went from around 20 seconds to < 1 second.

**Comment by garbit** [comment]
Experiencing the same issue of slow build / laggy hot-reload:

```
"nuxt": "^3.7.4"
```

The fix for me was to use 127.0.0.1:3000 and updating my run command to use `nuxt dev --host 127.0.0.1`.

**Comment by Lars-Apply** [comment]
this is still an issue - does Nuxt 3 not support 'localhost'?

**Comment by awssam** [comment]
this is a real problem i run this on my  machine
```
CPU
Intel Xeon-E 2386G - 6c/12t - 3.5 GHz/4.7 GHz
RAM
32 Go ECC 3200 MHz
Storage
512 Go SSD NVMe
```
taking almost 1 minutes to run the first request, can not move any further with this, with my laptop 2 minutes as well , even i am on my first steps on the project,

**Comment by MateuszZaniewski** [comment]
Im too having this problem. App takes about 30-40 sec to load, nitro 40sec. Any HMR is dead for now. Any change in text or style takes about 10-15 sec to load. Changing from localhost to IP does not help :/. Anyony have other solution?

**Comment by mklueh** [comment]
Same here.

ℹ Vite client warmed up in 27239ms
[nitro] ✔ Nitro built in 25354 ms
ℹ nuxt.config.ts updated. Restarting nuxt...
ℹ Vite server warmed up in 33279ms
ℹ ✨ optimized dependencies changed. reloading (x2)
ℹ Vite client warmed up in 41256ms
[nitro] ✔ Nitro built in 14183 ms

And the page is barely loading at all. 


I'm only using these plugins

```
  modules: [
    "@unocss/nuxt",
    "@nuxt/content",
    "@vueuse/nuxt",
    "nuxt-icon",
  ],
```

and it's just a really small blog that once was pretty fast in it's current state with the same dependencies.

**Comment by danielroe** [comment]
@mklueh It looks like you're using 3.8.1 which was meant to seriously improve this. Would you see if you can open a minimal reproduction or share access with me?

**Comment by MateuszZaniewski** [comment]
> Im too having this problem. App takes about 30-40 sec to load, nitro 40sec. Any HMR is dead for now. Any change in text or style takes about 10-15 sec to load. Changing from localhost to IP does not help :/. Anyony have other solution?

I manage to solve this issue. For me it was primevue combined with tailwind - unistall them too make HMR runs again, also vite reduced from 30-40 sec to 10. 

In my opinion it was some collision between styles (my css files, tailwind and primevue styles). First I tried removing all <style> tags from every file in my App and i helped to reduce time by half, but waiting 5-6sec to see any changes its stilk anoying. I hope this helps someone.

**Comment by existe-deja** [comment]
I just noticed the DOM inspector panel in firefox developer console is related to this. Other panels works fine, but the if the DOM inspector is open it's a nightmare

**Comment by rospirski** [comment]
> Experiencing the same issue of slow build / laggy hot-reload:
> 
> ```
> "nuxt": "^3.7.4"
> ```
> 
> The fix for me was to use `127.0.0.1:3000` and updating my run command to use `nuxt dev --host 127.0.0.1`.

Very nice, it worked for me
THANKSS

**Comment by ThomasFindlay** [comment]
The same issue still persists with Nuxt 3.8.1. On localhost, the requests to fetch all imports take ages and sometimes even result in a few reloads when Vite finds a new dependency. When serving the app via 127.0.0.1 it's much, much faster.

**Comment by Plinpod** [comment]
> --host 127.0.0.1

Identical issue for me on 3.8.1 

localhost is painfully slow, while 127.0.0.1 works fine. Does anyone have a clue why?

Environment is setup via WSL on Windows.

**Comment by ManuelOchagavia** [comment]
The same for me.  The environment is WSL. Using 127.0.0.1 changed the page load from ~50 secs to <1 sec

**Comment by AlansCodeLog** [comment]
I was having a similar issue, warmup + nitro build were taking forever. Turns out for me the problem was I had a some large directories in the folder (was trying out .direnv on nix + I had electron building to a non-standard dir .output-electron). After ignoring them times went down from 1 minute or more to just 5 seconds.

**Comment by tobychidi** [comment]
> I was having a similar issue, warmup + nitro build were taking forever. Turns out for me the problem was I had a some large directories in the folder (was trying out .direnv on nix + I had electron building to a non-standard dir .output-electron). After ignoring them times went down from 1 minute or more to just 5 seconds.

How did you implement the ignore? Starting my dev server takes ages. And Nuxt-ui is always being reported as a slow module. 

![image](https://github.com/nuxt/nuxt/assets/111127651/7194edd7-03b6-422c-b1a4-3099e6a284a7)
![image](https://github.com/nuxt/nuxt/assets/111127651/531b4e14-0500-48d1-9837-6dc25670b277)
![image](https://github.com/nuxt/nuxt/assets/111127651/0df572a9-8066-4563-adcc-b8847508a536)

**Comment by AlansCodeLog** [comment]
@tobychidi Sorry I didn't clarify. You can use .nuxtignore or the [ignore option](https://nuxt.com/docs/api/nuxt-config#ignore) in the config is the same thing.

Am using nuxt now for more complicated projects and putting everything in the root then scanning nearly everything by default is not a good default imo.

Personally I've switched to putting everything in src (using the srcDir option) to avoid any further issues. You can then do the following in the nuxt ignore:
```
./* # ignore everything
!./src/** # un-ignore src
```

**Comment by Neoniq** [comment]
Bump, localhost slow on windows+pnpm+ node20
Latest nuxt with freshly created project.
127.0.0.1 helps but only for page reload after first slow start.

Was migrating from vite where dev server starts in half a second, on nuxt it almost a minute

RIG: Ryzen 5800x3d 32gb + ssd

**Comment by bjoernfal** [comment]
Try using the --no-fork option!

________________________________

**Comment by gregmsanderson** [comment]
For anyone else getting this, the solution suggested by @AlansCodeLog helped for me.

It looks like Nuxt 4 will move over to that approach by default (though using `app/` rather than `src/`): https://nuxt.com/docs/getting-started/upgrade#new-directory-structure

**Comment by sethrw** [comment]
@tobychidi - Did you ever figure out the issue with nuxt ui? My page load time is 30+ seconds in dev and prod and it's almost certainly caused by nuxt ui. The above solutions did not help.

**Comment by imding** [comment]
![image](https://github.com/user-attachments/assets/44bc8ad8-96b8-410c-ba8d-8c34295c81b6)

Having extremely slow times here right after updating to 3.12.4 from 3.11.2.

**Comment by Hastatus12** [comment]
I'm having the same problem. Fresh and empty app takes more than 15 seconds to start. I'm using windows 10 with 12 GB RAM
```
√ Vite client built in 191ms                                                                                  11:03:27
√ Vite server built in 1141ms                                                                                 11:03:28
√ Nuxt Nitro server built in 2678 ms                                                                    nitro 11:03:32
i Vite client warmed up in 7ms                                                                                11:03:32
i Vite server warmed up in 1827ms                                                                             11:03:34
```

**Comment by conrado-l** [comment]
I'm still having this issue with 32 GB RAM using WSL2 with Ubuntu and also trying to run it natively on Windows 11, same result.

Tried both Nuxt 3.14 and 3.16, same result:

![Image](https://github.com/user-attachments/assets/d55350a4-ec0a-4c38-ad60-278bf7a8c538)

![Image](https://github.com/user-attachments/assets/41428cba-0141-4a89-948d-a577e36c4085)

This is unusable in terms of development experience.

**Comment by shaobeichen** [comment]
> The same issue still persists with Nuxt 3.8.1. On localhost, the requests to fetch all imports take ages and sometimes even result in a few reloads when Vite finds a new dependency. When serving the app via 127.0.0.1 it's much, much faster.同样的问题在 Nuxt 3.8.1 中仍然存在。在本地主机上，获取所有导入的请求需要很长时间，有时甚至会导致在 Vite 找到新的依赖项时重新加载一些。通过 127.0.0.1 提供应用程序时，速度要快得多。

This is useful for me

**Comment by tomsour1s** [comment]
I have exactly the same problem. I use nuxt 4 with the following settings:
```
  compatibilityDate: '2024-04-03',
  future: {
    compatibilityVersion: 4,
  },
```
My system is Windows with WSL2.
When I use localhost:3000 the loading times are extremely high. When I use 127.0.0.1 the load times are extremely low. Around a 10x of speed improvement
However, this leads to problems with my auth provider (OIDC).

**Comment by 50l3r** [comment]
The performance issue appears to be on Windows. It doesn't occur on Mac.

**Comment by tomsour1s** [comment]
I have a similar problem, but unlike described in the issue, the loading times for my project are extremely long when I do a “full page reload”. The loading times are then approx. 4-8 seconds. 
I use Nuxt4 on a Windows OS with WSL2. 
As described above, the problem also disappears for me if I change the dev server to 127.0.0.1.

**Comment by conrado-l** [comment]
I solved it. 

In my case I was using Vuetify with a template/theme, and I found out that using too much custom CSS slowed down the server warmup and initial request by a lot.

The fix was to **uninstall** the `sass` package and install `sass-embedded` instead, with these settings, which speed up SASS compilation tremendously:

`nuxt.config.ts`

```
vite: {
  css: {
    preprocessorOptions: {
      scss: {
        api: 'modern-compiler',
      } 
   }
}
```


References: 
- https://nuxt.vuetifyjs.com/guide/sass-customization.html
- https://vuetifyjs.com/en/features/sass-variables/#build-performance
- https://vite.dev/config/shared-options.html#css-preprocessoroptions
